### PR TITLE
Add Package.swift

### PR DESCRIPTION
**Pull request by ronaldmannak**
This pull request addresses the issue of projects manually copying the LLM directory from the Libraries folder.

With this PR, developers can now include the LLM and/or MNIST libraries in their projects via Swift Package Manager. This update simplifies the process of keeping the LLM dependency current.

### CODE REVIEW

**Comment by ronaldmannak** [comment]
As discussed in #31

**Comment by ronaldmannak** [comment]
It looks like lint fails, but I don't see any useful error. Who can help me with this?

**Comment by davidkoski** [comment]
> It looks like lint fails, but I don't see any useful error. Who can help me with this?

Sure, the ci builds says:

```
swift-format.............................................................Failed
```

It runs swift-format to make sure all the files have a consistent style -- this includes Package.swift!

Just run:

```
pre-commit run --all-files
```

**Comment by ronaldmannak** [comment]
@davidkoski Thanks but I understood that part, I was hoping to get a useful error message from the CI logs. 

It turned out to be not much work to install swift-format and add a pre-commit. Committed fix.

**Comment by ronaldmannak** [comment]
The targets are currently named MLXLLM and MLXMNIST, while the example libraries are simply called LLM and MNIST. If there's a preference for the simpler names, I'm open to rename the targets.

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// swift-tools-version: 5.9
+// The swift-tools-version declares the minimum version of Swift required to build this package.
+
+import PackageDescription
```
Comment (davidkoski) [inline Package.swift:4]: This is still failing the swift-format check.   In particular it wants:

```swift
 import PackageDescription
+
 let package = Package(
```

**Comment by davidkoski** [comment]
I think this is a really great idea!

Please add info to the top level README.md indicating how to make use of it.

File: Package.swift (hunk 1 above)
Comment (ronaldmannak) [reply Package.swift:4]: fixed in the latest commit. Thanks.

**Review by davidkoski — APPROVED** [review]
Looks great, thank you for the idea and the contribution!

**Pull request merged**